packages/components/src/ui/context/use-context-system.js

```diff
@@ -71,8 +71,13 @@ export function useContextSystem( props, namespace ) {
 		finalComponentProps[ key ] = overrideProps[ key ];
 	}
 
-	// @ts-ignore
-	finalComponentProps.children = rendered;
+	// Setting an `undefined` explicitly can cause unintended overwrites
+	// when a `cloneElement()` is involved.
+	if ( rendered !== undefined ) {
+		// @ts-ignore
+		finalComponentProps.children = rendered;
+	}
```

Comment:
In the very unlikely case that someone wants to explicitly wipe out the inherent children of a component using `cloneElement()`, they can still pass `children=null`.

For normal components that don't involve `cloneElement()`, `children` overrides will be (and have been) ignored anyway.